src/main/java/uk/gov/laa/ccms/caab/service/AssessmentService.java

```diff
@@ -0,0 +1,559 @@
+   * last saved date within their audit details.
+   *
+   * @param assessments the list of assessment details
+   * @return the most recent assessment detail, or null if the list is empty or null
+   */
+  public AssessmentDetail getMostRecentAssessmentDetail(final List<AssessmentDetail> assessments) {
+    return assessments != null ? assessments.stream()
+        .max(Comparator.comparing(assessment -> assessment.getAuditDetail().getLastSaved(),
+            Comparator.nullsFirst(Comparator.naturalOrder())))
+        .orElse(null) : null;
+  }
+
+  /**
+   * Calculates and updates the statuses for both means and merits assessments of an application.
+   * It determines which assessment was last saved to set the isMeansLast flag and applies status
+   * updates based on whether assessments have started or changed.
+   *
+   * @param application the application whose assessments are being evaluated
+   * @param meansAssessment the means assessment details
+   * @param meritsAssessment the merits assessment details
+   * @param user the user performing the update operation
+   */
+  public void calculateAssessmentStatuses(
+      final ApplicationDetail application,
+      final AssessmentDetail meansAssessment,
+      final AssessmentDetail meritsAssessment,
+      final UserDetail user) {
+    log.info("Calculating assessment statuses");
+
+    if (meansAssessment != null) {
+      if (application.getMeansAssessmentStatus() == null) {
+        if (!NOT_STARTED.getStatus().equals(meansAssessment.getStatus())) {
+          application.setMeansAssessmentAmended(true);
+        }
+      } else if (!application.getMeansAssessmentStatus().equals(meansAssessment.getStatus())) {
+        application.setMeansAssessmentAmended(true);
+      }
+    }
+
+    log.info("Calculating means assessment status");
+    calculateAssessmentStatus(meansAssessment, application, user);
+
+    log.info("Calculating merits assessment status");
+    calculateAssessmentStatus(meritsAssessment, application, user);
+  }
+
+  /**
+   * Calculates and updates the status of the provided assessment based on specific criteria,
+   * including the amendment state of the application and whether a reassessment is required.
+   * The status is updated through an API client and also set on the application object.
+   *
+   * @param currentAssessment the assessment whose status needs to be evaluated
+   * @param application the application related to the assessment
+   * @param user the user performing the update operation
+   */
+  private void calculateAssessmentStatus(
+      final AssessmentDetail currentAssessment,
+      final ApplicationDetail application,
+      final UserDetail user) {
+
+    boolean statusChanged = false;
+    String statusKey = getStatus(currentAssessment);
+
+    if (!application.getAmendment()
+        || (isApplicationsAssessmentAmended(application, currentAssessment))) {
+
+      if (COMPLETE.getStatus().equalsIgnoreCase(statusKey)
```

Comment:
If statusKey is an enum instance, then this comparison can be simplified to 
COMPLETE == statusKey || ERROR == statusKey

Reply:
Amended